fix(i18n,client): translatable cta url

naomi-lgbt opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->
Currently, a camper visiting the Chinese `/learn` page will see call to action links to help with the translation efforts. These links are hard-coded to point to the English version of the contributing documentation.

With this PR, those links are pulled out into the `translations.json` object so they can be replaced with a language-specific link to the appropriate internationalised version of our documentation.

Commit:
fix(i18n,client): translatable cta url

Makes the url for the i18n cta translatable instead of hard-coded.

RandellDawson commented:
This seems more like a config than a translation, so I am not sure `translations.json` is the best place for this.  I would think we would want tighter control over these links than to rely on Crowdin integration to get these right.

nhcarrigan commented:
That's a good point. 

Our footer links (for like the donate and about page) are also in the `translations.json` object. Should we extract those and compile a `links.json` object so we can control them through GitHub like we do the meta tags and trending news articles?

RandellDawson commented:
> That's a good point.
> 
> Our footer links (for like the donate and about page) are also in the `translations.json` object. Should we extract those and compile a `links.json` object so we can control them through GitHub like we do the meta tags and trending news articles?

That is what I am thinking.

nhcarrigan commented:
Will update the PR today when I have time

Commit:
feat: extract URL values

Extracts the URL values from the translations.json object and adds
them to a new links.json object. This will allow us to translate
these through GitHub instead, ensuring we have maximum control over
the values.

nhcarrigan commented:
Okay @RandellDawson I've extracted the URL values from `translations.json` and created a new `links.json` to keep off of Crowdin.

Commit:
fix: corrections for CI

Reverts changes to Chinese and Espanol translation.json objects,
regenerates snapshots for footer to pass tests.

nhcarrigan commented on client/i18n/locales/espanol/translations.json:
```diff
@@ -283,7 +283,7 @@
     "chal-preview": "Vista previa del desafío",
     "cert-map-estimates": {
-      "certs": "Certificación {{title}} (300 horas)",
+      "certs": "Certificación {{title}} (300 horas)",
```
Huh?? 👀

Commit:
fix: reset files

raisedadead reviewed (commented):
Code LGTM. 

I see commits were pushed after the ready for QA was added, just confirming if this is good to go, before I pull it down.

nhcarrigan commented:
This is good to go, yes

raisedadead commented:
@freeCodeCamp/dev-team please approve.

RandellDawson commented:
@nhcarrigan Great job on this PR!
